### ISSUE TITLE
Loinc 14292-7:  "IgE [Moles/volume] in Serum"

### ISSUE THREAD

**Issue by pnrobinson** [obophenotype/human-phenotype-ontology#3642]
Suggest creating a new term for Loinc: 14292-7 [IgE [Moles/volume] in Serum] 
New term label:
New term comment (if any):
Your biocurator ID for loinc2hpo (if desired): HPO:probinson

Create terms such as Abnormal IgE, etc level in blood.

**Comment by kingmanzhang** [comment]
Can you create this term and a child term, e.g. presence of IgE antibody for peanuts? I will spend some time to batch create a list of terms for "presence of IgE antibody for XXX" using your example as a template. @pnrobinson @nicolevasilevsky @LCCarmody

**Comment by pnrobinson** [comment]
https://hpo.jax.org/app/browse/term/HP:0003212
and
https://hpo.jax.org/app/browse/term/HP:0005479
-- currently there is no IgE parent term--maybe we should create these.

**Comment by pnrobinson** [comment]
Note that Increased IgE is not a good parent term for presence of anti peanut IgE etc --


Peter Robinson
Professor and Donald A. Roux Chair, Genomics and Computational Biology
The Jackson Laboratory for Genomic Medicine
860.837.2095 t | peter.robinson@jax.org | https://robinsongroup.github.io/
Peter Robinson
________________________________

**Comment by nicolevasilevsky** [comment]
@pnrobinson Are you proposing the new term Abnormal IgE level, to be a grouping class for Increased IgE level HP:0003212 and IgE deficiency HP:0005479?

Do we want to create granular terms for presence of anti-peanut IgE, etc., for all the allergens noted in LOINC?

What do you propose the parent class should be? Would Abnormal IgE level be an appropriate parent?

**Comment by LCCarmody** [comment]
HI All,
There IS a parent term: IgE-mediated food allergy (HP:0500093) because the immunologists requested it.  https://github.com/obophenotype/human-phenotype-ontology/issues/3425

This was added in April, so I guess the online HPO site is out of date? Not sure.

**Comment by pnrobinson** [comment]
The HPO website will be updated by tomorrow at the latest

**Comment by nicolevasilevsky** [comment]
should we map the LOINC terms to 'IgE-mediated food allergy'?

**Comment by kingmanzhang** [comment]
Could we prioritize this ticket? We will be able to have a few hundred more LOINC annotations instantly. 
I like Nicole's idea that we create a parent term for abnormal IgE in serum. We could map many LOINC tests to it as a temporary solution. @pnrobinson

**Comment by kingmanzhang** [comment]
This is the same list I sent out via email. 

List of antigens:
[IgE_antigens_for_HPO.tsv.zip](https://github.com/obophenotype/human-phenotype-ontology/files/2362434/IgE_antigens_for_HPO.tsv.zip)

**Comment by kingmanzhang** [comment]
@nicolevasilevsky I suggest you try to record the new HPO id for their corresponding LOINC in your template so that we can write a script to for annotation, instead of having to annotate manually one by one. 

Also be aware that there are duplications of antigens in the list. It happens when multiple LOINC tests are about the same kind of IgE.

**Comment by nicolevasilevsky** [comment]
@pnrobinson, @cmungall and @balhoff - is it possible to create a logical definition for terms such as 'increased anti-peanut IgE level'?
It would be something like a

'has part' some 
    ('increased amount'
     and ('inheres in' some 'immunoglobulin complex' and (**[not sure what object property to use here]** some %s)))
     and ('has modifier' some abnormal))

where % is the antigen, it could be a food, organism (such as a virus), plant (such as bamboo), etc.

Not sure if I am on the right track, any feedback is appreciated! 

Basically, an anti-peanut IgE is an IgE immunoglobulin complex that targets/recognizes specific epitopes in peanuts (the antigen)

**Comment by pnrobinson** [comment]
@nicolevasilevsky  Please do not go ahead just yet-- @kingmanzhang we need to extract the various patterns that occur in LOINC and in the literature. I am going to be with 20 immunologists in Vienna next week and I will also ask their advice about this!

**Comment by nicolevasilevsky** [comment]
@pnrobinson I won't do anything without your go ahead! :)

@kingmanzhang - maybe we could have a quick call on Monday and I can explain how this process works, it will be easier to show you via screen share

**Comment by kingmanzhang** [comment]
@pnrobinson What kind of patterns are you referring to? I think they almost exclusively fit the pattern "Elevated level of X IgE in serum". Are you suggesting classifying the antigens into classes, e.g. plants, animals etc? Regardless, I think potentially the bottleneck is to identify the antigens from other ontologies that we can make use of, such as the food ontology.

**Comment by nicolevasilevsky** [comment]
@balhoff there is [abnormalLevelOfMolecularEntity](https://github.com/obophenotype/upheno/blob/master/src/patterns/abnormalLevelOfMolecularEntity.yaml), I modeled the `abnormalLevelOfCellularComponent.yaml` after this one. That one uses CHEBI:23367 for  molecular entity. If we just changed that to entity and the label to abnormalLevelOfEntity, than we could combine these two, yes?

**Comment by cmungall** [comment]
I think you can use FOODON for any food antigen. We already import this in MONDO, for this DP: https://github.com/monarch-initiative/mondo/blob/master/src/patterns/allergy.yaml

it's not clear we need both the disease and phenotype here, happy to cede when you're done

**Comment by nicolevasilevsky** [comment]
@pnrobinson and @kingmanzhang: per our conversation at the hackathon last week, I skimmed through the list of antigens in the spreadsheet above.

I added the list to a [google doc](https://docs.google.com/spreadsheets/d/19fP64ZOdWw5iW2V2rcjONtHIQfSV7PHj7Ty01A7IYUg/edit#gid=1619558367) – in the tab **IgE_antigens_for_HPO Sorted**, this contains all the antigens (Ags) that Aaron pulled out from the LOINC file. In the tab **ClassificationOfAgs**, I came up with a list of 29 higher level classifications of most antigens in this list. @pnrobinson  – can you review this list and let me know what you think?  

I am sure I'm missing some Ags, but I think this list of 29 Ags would be a good list to start with, and we can create new terms either using the pattern above with DOSDP tools, or I can just manually add them, since there are not very many (pending @pnrobinson's approval), and then we can start annotating these terms in LOINC2HPO.

**Comment by LCCarmody** [comment]
Where do we stand on this? @pnrobinson Should we add terms?

**Comment by nicolevasilevsky** [comment]
I am about to meet with @kingmanzhang right now to discuss adding these terms via DOSDP tools

**Comment by pnrobinson** [comment]
I think we are on track. There is a new term, HP_0410241, Abnormal IgE level, that can be used for this LOINC term @kingmanzhang
